.github/workflows/ci.yaml

```diff
@@ -11,6 +11,18 @@ on:
     - cron:  '0 17 * * *'
 
 jobs:
+  shell-ci:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - uses: actions/checkout@v1
+    - name: Lint shell scripts, ignoring third-party files
+      run: |
+        find . | grep -e "\.sh$" | grep -v vendor | grep -v node_modules | grep -v demo/demo-magic > shell_files.out
```

Comment:
nit: I think you can use just the `find` command to match on the suffix name (instead of grep) and dir names.

Reply:
Done. Find can also negate a pattern match  by using ! -name "*vendor*" but I feel that grep is the clearer way to do this since find already does a positive match against *.sh.